distributed/core.py

```diff
@@ -672,3 +673,27 @@ def clean_exception(exception, traceback, **kwargs):
     elif isinstance(traceback, string_types):
         traceback = None  # happens if the traceback failed serializing
     return type(exception), exception, traceback
+
+
+def clean_loop(ioloop, new=False):
+    """ Normalize IOLoop to be Tornado compatible
+
+    If the loop does not exist, create it.
+    If the loop is an asyncio ioloop, then call converter codes
+    If new then start a new ioloop rather than use the global loop of this thread
+    """
+
+    if isinstance(ioloop, tornado.ioloop.IOLoop):
+        return ioloop
+    if not ioloop:
+        if new:
+            return tornado.ioloop.IOLoop()
+        else:
+            return tornado.ioloop.IOLoop.current()
+    try:
+        import asyncio
```

Comment:
I think it would be better not to import this unconditionally, as it can lengthen startup times. You could check for "asyncio" in `sys.modules`.